TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TKGame.BackEnd;
+using TKGame.Components.Interface;
+using TKGame.Content.Weapons;
+using TKGame.Players;
+using TKGame.Status_Effects;
+
+namespace TKGame.PowerUps.RelatedEntities
+{
+    public class Shocked : Entity
+    {
+        private float LifeTime { get; set; }
+        private float Duration { get; set; }
+        private float TickInterval { get; set; }
+        private float DamagePerTick { get; set; }
+        private float ElapsedTime { get; set; }
+
+        public Shocked(float lifeTime,
+                        float duration,
+                        float tickInterval,
+                        float damagePerTick,
+                        IPhysicsComponent physics_,
+                        IGraphicsComponent graphics_)
+        {
+            components = new Dictionary<ComponentType, List<IComponent>>
+            {
+                { ComponentType.Physics,        new List<IComponent> { physics_ } },
+                { ComponentType.Graphics,       new List<IComponent> { graphics_ } },
+            };
+
+            LifeTime = lifeTime;
+            Duration = duration;
+            TickInterval = tickInterval;
+            DamagePerTick = damagePerTick;
+
+            entityTexture = Art.ShockedTexture;
+
+            entityName = "Shocked"; // name for player class
+            entityType = EntityType.PowerUp;
+            Position = new Vector2(Player.Instance.HitBox.X - 150, Player.Instance.HitBox.Y);
+            HitBox = new Rectangle((int)Position.X, (int)Position.Y, 185, 150);
+            //Size = HitBox.Size;
+        }
+
+        public override void Update(GameTime gameTime)
+        {
+            ConfigureHitBox();
+            components[ComponentType.Physics].OfType<IPhysicsComponent>().First().Update(this, gameTime);
+
+            ElapsedTime += (float)gameTime.ElapsedGameTime.TotalSeconds;
+
+            if (ElapsedTime >= LifeTime)
+            {
+                this.IsExpired = true;
+                return;
+            }
+
+            List<Entity> entities = EntityManager.GetEntities();
+
+            foreach (Entity e in entities)
+            {
+                if (e != this && HitBox.Intersects(e.HitBox) && e != Player.Instance)
```

Comment:
Whenever I read these loops with hitbox detection I get confused if I read it wrong, we might want to use this. explicitly more often for situations like this

Reply:
I've used hitbox detection in loops also, we may just want to add comments to these lines to specify what we are doing exactly